gateway/mw_version_check_test.go

```diff
@@ -463,3 +464,57 @@ func TestOldVersioning_Expires(t *testing.T) {
 		})
 	})
 }
+
+func TestOldCache(t *testing.T) {
+	ts := StartTest(nil)
+	defer ts.Close()
+	var response = "default"
+
+	vInfo := apidef.VersionInfo{
+		ExtendedPaths: apidef.ExtendedPathsSet{
+			Cached: []string{
+				"/test",
+			},
+		},
+		UseExtendedPaths: true,
+	}
+
+	upstream := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(response))
+	}))
+
+	api := func() *APISpec {
+		return BuildAPI(func(spec *APISpec) {
+			spec.Proxy.ListenPath = "/"
+			spec.Proxy.TargetURL = upstream.URL
+			spec.VersionDefinition.Location = apidef.URLParamLocation
+			spec.VersionData.Versions = map[string]apidef.VersionInfo{
+				"Default": vInfo,
+			}
+			spec.CacheOptions = apidef.CacheOptions{
+				CacheTimeout: 120,
+				EnableCache:  true,
+			}
+		})[0]
+	}
+
+	check := func(t *testing.T, response string) {
+
+		_, _ = ts.Run(t, []test.TestCase{
+			{Path: "/test", BodyMatch: "default", Code: http.StatusOK},
+			{Path: "/anything", BodyMatch: response, Code: http.StatusOK},
+		}...)
+
+	}
+
+	ts.Gw.LoadAPI(api())
```

Comment:
I think load should be done inside check function. Right now,  migrated api is not loaded. Check function should get api param.